pkg/webui/console/containers/top-entities-dashboard-panel/item.js

```diff
@@ -12,18 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import React from 'react'
+import React, { useEffect } from 'react'
 import classNames from 'classnames'
+import { useDispatch, useSelector } from 'react-redux'
 
 import { Table } from '@ttn-lw/components/table'
 
 import useBookmark from '@ttn-lw/lib/hooks/use-bookmark'
 import PropTypes from '@ttn-lw/lib/prop-types'
 
+import { startGatewayStatistics, stopGatewayStatistics } from '@console/store/actions/gateways'
+
+import { selectDeviceLastSeen } from '@console/store/selectors/devices'
+import { selectApplicationDerivedLastSeen } from '@console/store/selectors/applications'
+import { selectGatewayStatistics } from '@console/store/selectors/gateways'
+import { selectGatewayLastSeen } from '@console/store/selectors/gateway-status'
+
 import styles from './top-entities-panel.styl'
 
 const EntitiesItem = ({ bookmark, headers, last }) => {
+  const dispatch = useDispatch()
   const { title, ids, path, icon } = useBookmark(bookmark)
+  const entityIds = bookmark.entity_ids
+  const entity = Object.keys(entityIds)[0].replace('_ids', '')
+  const deviceLastSeen = useSelector(state => selectDeviceLastSeen(state, ids.appId, ids.id))
+  const appLastSeen = useSelector(state => selectApplicationDerivedLastSeen(state, ids.id))
+  const statistics = useSelector(selectGatewayStatistics)
+  const gatewayLastSeen = useSelector(selectGatewayLastSeen)
+
+  useEffect(() => {
+    dispatch(startGatewayStatistics(ids.id))
+    return () => {
+      dispatch(stopGatewayStatistics())
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
This can only ever work for a single gateway. The list however has to retrieve the last seen of a range of gateways. I'm afraid, to implement this we'll have to add some new store logic which does this.
What I have in mind here is a logic that aggregates the IDs of gateways that we need the latest activity info on and then opens a combined stream for them which looks for ["heartbeat" events](https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.30/pkg/webui/console/containers/gateway-connection/use-connection-reactor.js#L36-L37) only. These heartbeat events are then used to update the lastSeen value for each gateway which we can then use to display in the table.

Reply:
Knowing that this would be a bigger task, let's use just the `Connected`/`Disconnected` info for now, like we do in the current gateway list and then we can add this later.